pkg/cpio/cpio.go

```diff
@@ -77,9 +78,28 @@ func (r Reader) ReadRecords() ([]Record, error) {
 
 type Writer struct {
 	rw RecordWriter
+	// There seems to be no harm done in stripping
+	// duplicate names when the record is written,
+	// and lots of harm done if we don't do it.
+	alreadyWritten map[string] bool
 }
 
 func (w Writer) WriteRecord(rec Record) error {
+	// we do NOT write records with absolute paths.
+	if filepath.IsAbs(rec.Name) {
```

Comment:
well, yes, but: you need to put the full path in GetRecord, since we have to defer opening until such time as we want to copy the file out in WriteRecord. So changing it in GetRecord will break the file reading step. Now, we could put the full path into the deferred open in GetRecord and the non-abs path in the Record. I'm kind of ok with that. BUT: there's no rule that says we have to ONLY take records from GetRecord. Should someone later add some new type of GetRecord function, or just assemble their own records and pass them to WriteRecord, well, they'd have to make sure not to generate absolute paths. Since WriteRecord is the gate through which all outgoing records must pass, it seems the best to enforce the restriction there.

I'm amenable to moving this enforcement to GetRecord but that approach does have its own set of nasties.